### ISSUE TITLE
Crash when running vg viz

### ISSUE THREAD

**Issue by Jorisvansteenbrugge** [vgteam/vg#3437]
**1. What were you trying to do?**
I was trying to run vg viz to produce an image of the graph

**2. What did you want to happen?**
to produce an output svg file

**3. What actually happened?**
The program crashed without producing the output

Copy-past from the terminal window:

```
vg: /home/xian/Documents/GitHub/source-tarball/vg/include/bdsg/internal/packed_structs.hpp:742: size_t bdsg::PackedDeque<Backend>::internal_index(const size_t&) const [with Backend = bdsg::STLBackend; size_t = long unsigned int]: Assertion `i < filled' failed.
ERROR: Signal 6 occurred. VG has crashed. Visit https://github.com/vgteam/vg/issues/new/choose to report a bug.
Stack trace path: /tmp/vg_crash_3xejmi/stacktrace.txt
Please include the stack trace file in your bug report!
```

**4. If you got a line like `Stack trace path: /somewhere/on/your/computer/stacktrace.txt`, please copy-paste the contents of that file here:**

```
Crash report for vg v1.35.0 "Ghizzano"
Stack trace (most recent call last):
#16   Object "/home/joris/tools/vg", at 0x5c85bd, in _start
#15   Object "/home/joris/tools/vg", at 0x1d720b1, in __libc_start_main
#14   Object "/home/joris/tools/vg", at 0xcc0e0b, in vg::subcommand::Subcommand::operator()(int, char**) const
#13   Object "/home/joris/tools/vg", at 0xce5cda, in main_viz(int, char**)
#12   Object "/home/joris/tools/vg", at 0xb7b6d8, in std::unique_ptr<handlegraph::PathHandleGraph, std::default_delete<handlegraph::PathHandleGraph> > vg::io::VPKG::load_one<handlegraph::PathHandleGraph>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
#11   Object "/home/joris/tools/vg", at 0xb7aafd, in std::_Function_handler<void (std::istream&), vg::io::VPKG::try_load_one<handlegraph::PathHandleGraph>(std::istream&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda(std::istream&)#1}>::_M_invoke(std::_Any_data const&, std::istream&)
#10   Object "/home/joris/tools/vg", at 0xb7bdc8, in std::_Function_handler<void (std::istream&), vg::io::VPKG::try_load_bare<handlegraph::PathHandleGraph>(std::istream&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)::{lambda(std::istream&)#1}>::_M_invoke(std::_Any_data const&, std::istream&)
#9    Object "/home/joris/tools/vg", at 0x132adc1, in std::_Function_handler<void* (std::istream&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&), vg::io::register_loader_saver_gfa()::{lambda(std::istream&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)#1}>::_M_invoke(std::_Any_data const&, std::istream&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
#8    Object "/home/joris/tools/vg", at 0x12c3f47, in vg::algorithms::gfa_to_path_handle_graph(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, handlegraph::MutablePathMutableHandleGraph*, bool, bool, long, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
#7    Object "/home/joris/tools/vg", at 0x12c2fe4, in vg::algorithms::gfa_to_handle_graph_add_paths(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::istream*, handlegraph::MutablePathHandleGraph*, gfak::GFAKluge&, vg::algorithms::IDMapInfo&)
#6    Object "/home/joris/tools/vg", at 0x12c53d9, in std::_Function_handler<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool), vg::algorithms::gfa_to_handle_graph_add_paths(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::istream*, handlegraph::MutablePathHandleGraph*, gfak::GFAKluge&, vg::algorithms::IDMapInfo&)::{lambda(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool)#1}>::_M_invoke(std::_Any_data const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::_Any_data const&, std::_Any_data const&)
#5    Object "/home/joris/tools/vg", at 0x6d4652, in bdsg::BasePackedGraph<bdsg::STLBackend>::append_step(handlegraph::path_handle_t const&, handlegraph::handle_t const&)
#4    Object "/home/joris/tools/vg", at 0x6b54a8, in bdsg::PackedDeque<bdsg::STLBackend>::get(unsigned long const&) const [clone .isra.0]
#3    Object "/home/joris/tools/vg", at 0x1d823f5, in __assert_fail
#2    Object "/home/joris/tools/vg", at 0x59c594, in __assert_fail_base.cold
#1    Object "/home/joris/tools/vg", at 0x59c6af, in abort
#0    Object "/home/joris/tools/vg", at 0x1424cbb, in raise
```

**5. What data and command can the vg dev team use to make the problem happen?**
`/home/joris/tools/vg viz -x MAF.GFA1 -o test.svg`
The file is available here: https://drive.google.com/file/d/1PUmXkqRipHbjgcURkcxAUqLMgkPNQqt4/view?usp=sharing

**6. What does running `vg version` say?**

```
vg version v1.35.0 "Ghizzano"
Compiled with g++ (Ubuntu 10.3.0-1ubuntu1) 10.3.0 on Linux
Linked against libstd++ 20210408
Built by xian@sept
```

**Comment by adamnovak** [comment]
This looks a little bit like maybe the GFA is visiting nodes that aren't really there; we aren't able to import it into our internal graph format, to even begin to draw it, and that problem arises when we're looking at the paths.

**Comment by adamnovak** [comment]
This looks to be a `gfakluge` bug; `for_each_path_element_in_file` is telling us we have a node called `"\nS"` on a path, because it's apparently reading off the end of the path line and trying to eat the next node line.

Putting the path as the last line in the file might work around it; I will see about fixing it,

**Comment by adamnovak** [comment]
OK, `gfakluge` doesn't really help us here with error checking, but the real problem is that the `P` line for `Mchit_Scaffold9` has only 3 fields in it, while [the GFA 1.0 spec requires 4 fields on the line](https://gfa-spec.github.io/GFA-spec/GFA1.html#required-fields-5). The "Overlaps" field is missing; you don't have to *actually* specify any overlaps, but if you don't you need a `*` in that last field.

The parser [just decides all of the file up to the next tab is the nodes that the path visits, ignoring the newline at the end of the path line](https://github.com/vgteam/gfakluge/blob/8f17321c6218a4a4e431acfca87f4bab0fc35546/src/gfakluge.hpp#L990-L993), so if that final field is missing we don't recognize it as a missing field and we just mis-parse the file instead.

**Comment by adamnovak** [comment]
OK, I've opened #3442 which I think will fix this issue, to the extent that vg can. When I run my new code on this GFA file I get an error like this:

```
Missing overlaps for GFA path "Mchit_Scaffold9"
```

That GFA path is indeed missing the required last field on the P line for that path, so I think the error checker is working. To actually load the file and run `vg viz`, @Jorisvansteenbrugge, you'll have to fix it up to match the GFA 1.0 specification first.

**Comment by Jorisvansteenbrugge** [comment]
Thank you for helping me identify the problem @adamnovak! In that case I think the software that I use upstream to generate the GFA file is not producing an entirely correct file. I will try to see if I can fix it over there.
